Flaky-test: ResourceGroupRateLimiterTest.testResourceGroupPublishRateLimit

### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.


### Example failure

https://github.com/apache/pulsar/actions/runs/5067587816/jobs/9099067011?pr=20384#step:10:1264

### Exception stacktrace

<!-- copy-paste the stack trace in the code block below -->
```
  Error:  Tests run: 6, Failures: 1, Errors: 0, Skipped: 5, Time elapsed: 21.799 s <<< FAILURE! - in org.apache.pulsar.broker.resourcegroup.ResourceGroupRateLimiterTest
  Error:  org.apache.pulsar.broker.resourcegroup.ResourceGroupRateLimiterTest.testResourceGroupPublishRateLimit  Time elapsed: 7.051 s  <<< FAILURE!
  org.awaitility.core.ConditionTimeoutException: Assertion condition defined as a org.apache.pulsar.broker.resourcegroup.ResourceGroupRateLimiterTest expected [null] but found [org.apache.pulsar.broker.resourcegroup.ResourceGroup@464bf589] within 1 seconds.
  	at org.awaitility.core.ConditionAwaiter.await(ConditionAwaiter.java:167)
  	at org.awaitility.core.AssertionCondition.await(AssertionCondition.java:119)
  	at org.awaitility.core.AssertionCondition.await(AssertionCondition.java:31)
  	at org.awaitility.core.ConditionFactory.until(ConditionFactory.java:985)
  	at org.awaitility.core.ConditionFactory.untilAsserted(ConditionFactory.java:769)
  	at org.apache.pulsar.broker.resourcegroup.ResourceGroupRateLimiterTest.deleteResourceGroup(ResourceGroupRateLimiterTest.java:80)
  	at org.apache.pulsar.broker.resourcegroup.ResourceGroupRateLimiterTest.testRateLimit(ResourceGroupRateLimiterTest.java:134)
  	at org.apache.pulsar.broker.resourcegroup.ResourceGroupRateLimiterTest.testResourceGroupPublishRateLimit(ResourceGroupRateLimiterTest.java:147)
  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
  	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
  	at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
  	at org.testng.internal.invokers.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:47)
  	at org.testng.internal.invokers.InvokeMethodRunnable.call(InvokeMethodRunnable.java:76)
  	at org.testng.internal.invokers.InvokeMethodRunnable.call(InvokeMethodRunnable.java:11)
  	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
  	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
  	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
  	at java.base/java.lang.Thread.run(Thread.java:833)
  Caused by: java.lang.AssertionError: expected [null] but found [org.apache.pulsar.broker.resourcegroup.ResourceGroup@464bf589]
  	at org.testng.Assert.fail(Assert.java:110)
  	at org.testng.Assert.failNotSame(Assert.java:1573)
  	at org.testng.Assert.assertNull(Assert.java:1506)
  	at org.testng.Assert.assertNull(Assert.java:1494)
  	at org.apache.pulsar.broker.resourcegroup.ResourceGroupRateLimiterTest.lambda$deleteResourceGroup$1(ResourceGroupRateLimiterTest.java:80)
  	at org.awaitility.core.AssertionCondition.lambda$new$0(AssertionCondition.java:53)
  	at org.awaitility.core.ConditionAwaiter$ConditionPoller.call(ConditionAwaiter.java:248)
  	at org.awaitility.core.ConditionAwaiter$ConditionPoller.call(ConditionAwaiter.java:235)
  	... 4 more
```


### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

--- a/pulsar-broker/src/test/java/org/apache/pulsar/broker/resourcegroup/ResourceGroupRateLimiterTest.java
+++ b/pulsar-broker/src/test/java/org/apache/pulsar/broker/resourcegroup/ResourceGroupRateLimiterTest.java
@@ -76,7 +76,7 @@ public class ResourceGroupRateLimiterTest extends BrokerTestBase {
 
     public void deleteResourceGroup(String rgName) throws PulsarAdminException {
         admin.resourcegroups().deleteResourceGroup(rgName);
-        Awaitility.await().atMost(1, TimeUnit.SECONDS)
+        Awaitility.await().atMost(10, TimeUnit.SECONDS)
             .untilAsserted(() -> assertNull(pulsar.getResourceGroupServiceManager().resourceGroupGet(rgName)));
     }
 

